main.go

```diff
@@ -36,12 +51,51 @@ type Firehose struct {
 	processSettings ProcessSettings
 	storageBackend  StorageBackend
 	hedwigConsumer  *hedwig.QueueConsumer
+	messageBuffer []*hedwig.Message
+	flushLock sync.Mutex
+	flushCh chan struct{}
+}
+
+func (fp *Firehose) WriteMessage(wg *sync.WaitGroup, message *hedwig.Message) error {
+	defer wg.Done()
+	// TODO: serialization for hedwig.Message to []byte
+	var data []byte
```

Comment:
should this be the `8 bytes for timestamp, 8 bytes for length of encoded message, n bytes for encoded message` how do I go from hedwig.Message to bytes?

Reply:
there's https://github.com/cloudchacho/hedwig-python/blob/1e4ca5472fe661ffd9d3cedd10a9ddc2daa0926b/hedwig/validators/base.py#L154-L181 and https://github.com/cloudchacho/hedwig-python/blob/1e4ca5472fe661ffd9d3cedd10a9ddc2daa0926b/hedwig/validators/base.py#L231-L249 .. seems like we'd need similar functions in go lib.